### PR TITLE
app-emulation/wine-vanilla, app-emulation/wine-staging, app-emulation/wine-proton: add comment how to fix linking with user patches

### PR DESCRIPTION
**Pull request by MrDuartePT**
When using adding some user patches to wine for exemple winehags is needed to run `tools/make_specfiles` to fix linking on the end of the compile (more info here: https://github.com/ValveSoftware/Proton/issues/7361#issuecomment-1863639235).

I don't know if other patches requires this but the compile also succeeds with line added (wine-staging compile with hags patches [still compiling]):
image

Also wine-proton is compiling just fine with the new line (but I have no patches since need to be updated to wine 9.0 for hags patches to work).

### CODE REVIEW

**Comment by ionenwks** [comment]
Not sure if I really want to add more of these than already are (esp. considering most users don't do patches or do patches that don't need this), I even considered removing the one already had but left it alone to avoid breaking *current* patch usage.

Note that an alternative for user patches is to have e.g.:
```
mkdir -p /etc/portage/env/app-emulation &&
cat > /etc/portage/env/app-emulation/wine-staging <<EOF
post_src_prepare() {
	tools/make_specfiles || die
	# whatever else the patch needs, e.g. make_vulkan
	# which is not run for non-proton at the moment
}
EOF
```
fwiw this one isn't as bad as make_vulkan given it doesn't require python eclasses (previously refused in https://bugs.gentoo.org/871960), but well

**Comment by ionenwks** [comment]
Guess I could add an ebuild comment to make that option clearer though, e.g.
```
    # always update for patches (including user's wrt #432348)
    eautoreconf
    tools/make_requests || die # perl
    # tip: if need more for user patches, with portage can e.g. do
    # echo "post_src_prepare() { tools/make_specfiles || die; }" \
    #     > /etc/portage/env/app-emulation/wine-vanilla
```

**Comment by MrDuartePT** [comment]
> Guess I could add an ebuild comment to make that option clearer though, e.g.
> 
> ```
>     # always update for patches (including user's wrt #432348)
>     eautoreconf
>     tools/make_requests || die # perl
>     # tip: if need more for user patches, with portage can e.g. do
>     # echo "post_src_prepare() { tools/make_specfiles || die; }" \
>     #     > /etc/portage/env/app-emulation/wine-vanilla
> ```

I can agree since make_vulkan create problem in the past we can add this one as a comment [make_specfile only recreate them but it better to be safe that sorry]

**Comment by MrDuartePT** [comment]
@ionenwks I think is ok. I added also wine-vanilla forget that one

File: app-emulation/wine-proton/wine-proton-7.0.6.ebuild [hunk 1]
```diff
@@ -178,6 +178,9 @@ src_prepare() {
 	eautoreconf
 	tools/make_requests || die # perl
 	dlls/winevulkan/make_vulkan -x vk.xml || die # python, needed for proton's
+	# tip: if need more for user patches, with portage can e.g. do
+    # echo "post_src_prepare() { tools/make_specfiles || die; }" \
+    #     > /etc/portage/env/app-emulation/wine-proton
```
Comment (ionenwks) [inline app-emulation/wine-proton/wine-proton-7.0.6.ebuild:183]:
```suggestion
	# echo "post_src_prepare() { tools/make_specfiles || die; }" \
	#     > /etc/portage/env/app-emulation/wine-proton
```
(same for the others, has whitespaces before `#`)

**Comment by ionenwks** [comment]
Well, I had a local draft for it (was thinking of adding the comment myself before you mentioned updating the PR), so I'll just go ahead and merge that.

**Pull request closed without merging**